packages/api/validations/workspace.js

```diff
@@ -0,0 +1,20 @@
+import * as yup from 'yup'
+import { validateDTO } from './index'
+
+const diseaseSchema = yup.object({
+  name: yup.string().min(1).max(512).required(),
+  abbreviation: yup.string(),
+  omimId: yup.string(),
+  causalGene: yup.string(),
+  mutationImpact: yup.string(),
+  proteinSize: yup.number().positive().min(1),
+})
+const workspaceSchema = yup.object().shape({
+  name: yup.string().min(1).max(256).required(),
+  description: yup.string().min(1).max(4096),
+  disease: diseaseSchema,
+})
+
+export const validateWorkspaceDto = (dto) => {
+  validateDTO(dto, workspaceSchema)
+}
```

Comment:
This is great. What benefits does it have over JSON Schema?

Reply:
Both of them as schema based (declarative) for validating the structure of JSON data, JSON Schema is spec for validation and the it's JS implementations are not so great compared to **Joi** or **Yup**  also JSON Schema  does not offer all the features you can get from Yup Like: Casting, adjust the schema based on a sibling or sibling children fields ...